src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -368,14 +369,55 @@ private String buildLookupEndpoint(String type, Object... params) {
     CompletableFuture<Void> statusFuture = getEntryId(INSTANCE_STATUSES, MISSING_INSTANCE_STATUS)
       .thenAccept(lookupObj::mergeIn);
 
-    CompletableFuture<Void> contributorNameTypeIdFuture = getEntryId(CONTRIBUTOR_NAME_TYPES, MISSING_CONTRIBUTOR_NAME_TYPE)
-      .thenAccept(lookupObj::mergeIn);
+    CompletableFuture<Void> contributorNameTypeIdFuture = verifyContributorNameTypesExist(compPOL.getContributors());
 
     return allOf(ctx, instanceTypeFuture, statusFuture, contributorNameTypeIdFuture)
       .thenApply(v -> buildInstanceRecordJsonObject(compPOL, lookupObj))
       .thenCompose(instanceRecJson -> createRecordInStorage(instanceRecJson, String.format(CREATE_INSTANCE_ENDPOINT, lang)));
   }
 
+  private CompletableFuture<Void> verifyContributorNameTypesExist(List<Contributor> contributors) {
+    List<String> ids = contributors.stream()
+      .map(Contributor::getContributorNameTypeId)
+      .distinct()
+      .collect(toList());
+
+    return getContributorNameTypes(ids)
+      .thenAccept(contributorNameTypes -> {
+
+        List<String> retrievedIds = contributorNameTypes.stream()
+          .map(o -> o.getString(ID).toLowerCase())
```

Comment:
is `toLowerCase` really required?

Reply:
I wanted to guarantee that `contributorNameTypeIds` from the inventory-storage will be in the same case as from the request. I forgot to lowercase `contributorNameTypeIds` from the request.